common/string_utils.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2020-2022 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ *
+ * This software product is a proprietary product of Nvidia Corporation and its affiliates
+ * (the "Company") and all right, title, and interest in and to the software
+ * product, including all associated intellectual property rights, are and
+ * shall remain exclusively with the Company.
+ *
+ * This software product is governed by the End User License Agreement
+ * provided with the software product.
+ */
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <stdio.h>
+#include <string.h>
+#include <memory>
+#include <cctype>
+#include <algorithm>
+#include <string>
+#include "string_utils.h"
+
+namespace string_utils
+{
+    std::string& trim(std::string& s)
```

Comment:
Considering the purpose of your modification, the intention is to replace the usage of "boost::algorithm::trim" with a different implementation.
The code you provided does not achieve the same functionality as "boost::algorithm::trim".
While "boost::algorithm::trim" removes all leading and trailing spaces from the input string, the proposed alternative "string_utils::trim" removes all spaces from the input, including those within the string.

Please also keep the original signature "void trim(std::string& Input)